pkg/controller/controller_instance.go

```diff
@@ -428,19 +447,22 @@ func (c *controller) reconcileServiceInstance(instance *v1alpha1.ServiceInstance
 
 	glog.V(4).Infof("Adding/Updating ServiceInstance %v/%v", instance.Namespace, instance.Name)
 
-	serviceClass, servicePlan, brokerName, brokerClient, err := c.getServiceClassPlanAndServiceBroker(instance)
-	if err != nil {
-		return err
-	}
-
 	// we will definitely update the instance's status - make a deep copy now
 	// for use later in this method.
 	clone, err := api.Scheme.DeepCopy(instance)
 	if err != nil {
-		return err
+		return false, err
 	}
 	toUpdate := clone.(*v1alpha1.ServiceInstance)
 
+	c.setServiceInstanceForPendingChange(toUpdate)
+	defer c.updateServiceInstanceStatus(toUpdate)
```

Comment:
I suspect deferring is going to be a problem for orphan mitigation. One of the situations that gets us into trouble is when our request to the broker completed, but our API server resource update failed.

I think for now it's fine, so you don't have to remove it from the PR. Just something we'll need to keep in mind when tackling orphan mitigation in the near future.

Reply:
Would it be sufficient to call updateServiceInstanceStatus just prior to making the call to the Broker to cover this case?

Reply:
We're going to have to update the status of the resource after contacting the broker, as the result of that operation informs what we want the eventual status to contain, so I wouldn't put an update before calling the broker.

On second thought, I'll need to do some more thinking about the exact semantics of orphan mitigation, so for the time being I wouldn't worry about the defer. Just something that came to mind as I was reviewing.